compiler/passes/scopeResolve.cpp

```diff
@@ -2404,6 +2404,18 @@ static void removeUnusedModules() {
   }
 }
 
+static void detectUserDefinedBorrowMethods() {
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    if (fn->isMethod()) {
+      if (strncmp(fn->name, "borrow", 6) == 0) {
```

Comment:
Won't this flag any method whose name starts with `"borrow"`?

Reply:
It's better to not use `strcmp` or `strncmp` in the compiler if you can avoid it. In this case, you can just do this:

``` c++
const char astrBorrow = astr("borrow");
forv_Vec(FnSymbol, fn, gFnSymbols) {
    if (fn->isMethod() && fn->name == astrBorrow) {
```